src/api/doAttachApp.js

```diff
@@ -60,6 +60,8 @@ export const doAttachApp = async (values, formApi, authenticationInitialValues,
 
         if (filteredValues.source && !isEmpty(filteredValues.source)) {
             promises.push(getSourcesApi().updateSource(sourceId, filteredValues.source));
+        } else {
+            promises.push(Promise.resolve(undefined));
```

Comment:
what is the purpose of these? Do you just need the correct order of arguments in promise.all?

Reply:
I need to extract info from responses, so I have to know indexes of all promises. And because they are dynamically added, I have to fix their positions by using `undefined`